pallets/exchange/src/lib.rs

```diff
@@ -294,6 +314,7 @@ impl<T: Config> Module<T> {
 	/// satisfying  that sum( sell_b_intentions.amount_sell ) <= sell_a_intention.amount_sell
 	///
 	/// Intention A must be valid - that means that it is verified first by validating if it was possible to do AMM trade.
+	// REVIEW: this function does not have tests, but should
```

Comment:
Although not directly called in the tests - this function is tested as part of on_finalize calls in the tests

Reply:
Your call to make. IMO this would make for a good target for property testing ([proptest](https://altsysrq.github.io/proptest-book/intro.html) and friends).